### PR TITLE
Courtesy PR

### PR DESCRIPTION
**Pull request by stigzler**
Hi,
I thought I'd ping over the latest version of this (I got carried away!). You'll see loads of new features, however, I doubt my coding is up to your standards and it certainly doesn't follow WPF/MVVM at all. I'm suspecting that you probably won't want to merge - totally understandable! I haven't really got time to refactor lots of this code, sadly. Anyways, let me know either way. 
Steve

### CODE REVIEW

**Comment by conwid** [comment]
Thanks for all the features, they seem very interesting. I do agree that the code could you some rewriting and as is I don't really feel comfortable merging, but I do see a lot of value in the PR and with a little work I think I can restructure it so that it is easier to manage. I'm also a little busy with other stuff, but I'll try to take the time soon and somehow integrate your code into my codebase!

**Comment by conwid** [comment]
I had some time during the holidays to look into your code a bit more. First of all, I have to say that you implemented some very cool features, so thank you. Unfortunately, I will have to reject this PR for a couple of reasons (and by no means are all of these on you):

- I am very uncomfortable with the Syncfusion library. I'm not sure about the licensing, and even if that is permissive enough, I'm not really fond of their products.
- Using the registry is also a bit fishy.
- This would only support a dark theme, not all VS themes.
- Lastly, the code needs a lot of restructuring regarding MVVM and just general software architecture and best practices. In its current form, it is quite hard to follow the code.

But thank you for taking the time to implement all these.

**Comment by stigzler** [comment]
No worries - I understand. I went ahead and developed my own version anyway. Best of luck to you. 👍
